get_github_coderabbit_review/coderabbit_fetcher/validation.py

```
@@ -0,0 +1,668 @@
+        def wrapper(*args, **kwargs):
+            last_exception = None
+
+            for attempt in range(max_attempts):
+                try:
+                    return func(*args, **kwargs)
+                except exceptions as e:
+                    last_exception = e
+
+                    if attempt == max_attempts - 1:
+                        # Last attempt failed, re-raise
+                        logger.error(f"Function {func.__name__} failed after {max_attempts} attempts: {e}")
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)